src/app/model-download/components/model-download-component/model-download-component.component.ts

```diff
@@ -0,0 +1,172 @@
+    private routeService: RouteService,
+    private informationSourceApolloService: WeakSourceApolloService,
+    private formBuilder: FormBuilder
+  ) { }
+
+  ngOnInit(): void {
+    this.initForm();
+    this.routeService.updateActivatedRoute(this.activatedRoute);
+    this.projectId = this.activatedRoute.parent.snapshot.paramMap.get('projectId');
+    [this.projectQuery$, this.project$] = this.projectApolloService.getProjectByIdQuery(this.projectId);
+    [this.downloadedModelsQuery$, this.downloadedModelsList$] = this.informationSourceApolloService.getModelProviderInfo();
+    this.prepareModels();
+    this.prepareZeroShotRecommendations(this.projectId);
+    this.isManaged = ConfigManager.getIsManaged();
+
+    this.subscriptions$.push(
+      this.downloadedModelsList$.subscribe((downloadedModels) => this.downloadedModels = downloadedModels));
+
+    NotificationService.subscribeToNotification(this, {
+      projectId: this.projectId,
+      whitelist: ['model_provider_download'],
+      func: this.handleWebsocketNotification
+    });
+  }
+
+  ngOnDestroy(): void {
+    this.subscriptions$.forEach((subscription) => subscription.unsubscribe());
+    NotificationService.unsubscribeFromNotification(this, this.projectId)
+  }
+
+  initForm() {
+    this.form = this.formBuilder.group({
+      name: ['', Validators.required]
+    })
+  }
+
+  deleteModel(name: string, revision: string) {
+    this.subscriptions$.push(
+      this.informationSourceApolloService
+      .deleteModel(name,revision)
+      .pipe(first()).subscribe()
+    );
+  }
+
+  downloadModel() {
+    if(this.form.invalid) return ;
+    this.subscriptions$.push(
+      this.informationSourceApolloService
+      .downloadModel(this.projectId, this.form.get('name').value)
+      .pipe(first()).subscribe()
+    );
+  }
+
+  parseUTC(utc: any) {
+    const milliseconds = +utc * 1000;
+    const utcDate = dateAsUTCDate(new Date(milliseconds));
+    return utcDate.toLocaleString();
+  }
+
+  prepareModels() {
+    this.subscriptions$.push(this.projectApolloService.getRecomendedEncodersForEmbeddings(this.projectId)
+      .subscribe((models) => {
+        this.models = models.filter(el =>
+          el.configString != 'bag-of-characters' && el.configString != 'bag-of-words' && el.configString != 'tf-idf');
+      }));
+  }
+
+  prepareZeroShotRecommendations(projectId: string) {
+    let q, vc;
+    [q, vc] = this.informationSourceApolloService.getZeroShotRecommendations(projectId);
+    this.subscriptions$.push(vc.pipe(first()).subscribe((r) => {
+      r.forEach((rec) => this.models.push(rec))
+    }));
+  }
+
+  formatBytes(bytes, decimals = 2) {
+    if (bytes === 0) return '0 Bytes';
+    const dm = decimals < 0 ? 0 : decimals;
+    const sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'];
+    const i = Math.floor(Math.log(bytes) / Math.log(1024));
+    return parseFloat((bytes / Math.pow(1024, i)).toFixed(dm)) + ' ' + sizes[i];
+  }
```

Comment:
can be put into helper-functions file (like dateAsUTCDate) and imported & used so changes only need to be done at one place